### PR TITLE
Feature/rest history integration tests

### PR DESCRIPTION
**Pull request by tiholic**
Adding integration tests for Rest History

1. Test history retrieval with default params
2. Test `pagination`
3. Test with non-default `direction`
4. Test with `start`
5. Test with `start` and `end`

### CODE REVIEW

File: test_integration/lib/test/rest_history_test.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    final historyLimit2 = await _history(channel, RestHistoryParams(limit: 2));
+    await Future.delayed(const Duration(seconds: 2));
+
+    final historyForwardLimit4 = await _history(
+        channel, RestHistoryParams(direction: 'forwards', limit: 4));
+    await Future.delayed(const Duration(seconds: 2));
+
+    final time1 = DateTime.now();
+    //TODO(tiholic) iOS fails without this delay
```
Comment (QuintinWillison) [inline test_integration/lib/test/rest_history_test.dart:49]: Could you please ensure there's an issue for these `TODO` comments relating to the delays required so we can keep track of this technical debt.

File: test_integration/lib/test/test_names.dart [hunk 2]
```diff
@@ -11,8 +11,10 @@ class TestName {
       'testHelperUnhandledExceptionTest';
 
   static const String restPublish = 'restPublish';
+  static const String restHistory = 'restHistory';
   static const String restPublishWithAuthCallback =
       'restPublishWithAuthCallback';
+  // TODO(tiholic) handle restHistoryWithAuthCallback
```
Comment (QuintinWillison) [inline test_integration/lib/test/test_names.dart:17]: Again, it looks like we should manage this deferred work with an issue.

**Review by QuintinWillison — APPROVED** [review]
I've not looked at the test implementations in detail for lack of time, however addition of any tests will always be better than having no tests so I'll trust in the checks passing. 😁

File: test_integration/lib/test/rest_history_test.dart (hunk 1 above)
Comment (tiholic) [reply test_integration/lib/test/rest_history_test.dart:49]: https://github.com/ably/ably-flutter/issues/86

File: test_integration/lib/test/test_names.dart (hunk 2 above)
Comment (tiholic) [reply test_integration/lib/test/test_names.dart:17]: https://github.com/ably/ably-flutter/issues/86

Comment (tiholic) [reply test_integration/lib/test/test_names.dart:17]: https://github.com/ably/ably-flutter/issues/87

**Pull request merged**